frontend/assignments/3/frontend/src/components/stockpage/GlobalHistory.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import Container from '@mui/material/Container';
+import { Box, Typography } from '@mui/material';
+import { FlexDivContainer } from '../utils/ui/layout/FlexDivContainer';
+import { SocketProps } from "../types/SocketProps";
+import { HistoryItem, TransactionType } from '../types/HistoryItem';
+
+export function GlobalHistory({ socket }: SocketProps) {
+    const [transactions, setTransactions] = useState<HistoryItem[]>([]);
+    const [username, setUsername] = useState<string>('');
+    const [symbol, setSymbol] = useState<string>('');
+    useEffect(() => {
+        socket?.on('global', (username: string, data: any,symbol:string) => {
+            const transaction = data as HistoryItem;
+            setUsername(username);
+            setSymbol(symbol);
+            if (transaction) {
+                setTransactions(prevTransactions => [...prevTransactions, transaction]);
+            }
+        });
+
+        return () => {
+            socket?.off('global');
```

Comment:
lets not do this check socket is not null or undefined then do the action. If they are null then log error in console.